src/shared/components/fusionTable/FusionTable.tsx

```diff
@@ -0,0 +1,335 @@
+
+            _renderColumnFn = (d: StructuralVariantExt[]) => {
+                const sampleId = d[0][label.attribute];
+                if (!molecularProfileIdToMolecularProfile || !studyIdToStudy)
+                    return <span/>;
+
+                const molecularProfileId = d[0].studyId + '_structural_variants';
+                const geneticProfile = molecularProfileIdToMolecularProfile[molecularProfileId];
+
+
+                const study = studyIdToStudy[geneticProfile.studyId];
+                let linkToPatientView: string = `#/patient?sampleId=${sampleId}&studyId=${study.studyId}`;
+
+                /**
+                 * HACK to deal with having mutation mapper on index.do
+                 * Change it to case.do
+                 * https://github.com/cBioPortal/cbioportal/issues/2783
+                 */
+                const indexLocation: number = window.location.href.search('index.do');
+
+                if (indexLocation > -1) {
+                    linkToPatientView = window.location.href.substring(0, indexLocation) + 'case.do' + linkToPatientView;
+                }
+                // END HACK
+
+                return <a href={linkToPatientView} target='_blank'>{sampleId}</a>
+            }
+        }
+
+        if (label.columnType === FusionTableColumnType.STUDY && this.props.studyIdToStudy) {
+            _renderColumnFn = (d: StructuralVariant[]) => {
+                if (!molecularProfileIdToMolecularProfile || !studyIdToStudy)
+                    return <span/>;
+
+                const molecularProfileId = d[0].studyId + '_structural_variants';
+                const geneticProfile = molecularProfileIdToMolecularProfile[molecularProfileId];
+
+                if (!geneticProfile)
+                    return <span/>;
+
+                const study = studyIdToStudy[geneticProfile.studyId];
+                return (
+                    <a href={getStudySummaryUrl(study.studyId)} target="_blank">
+                        <TruncatedText
+                            text={study.name}
+                            tooltip={<div style={{maxWidth: 300}} dangerouslySetInnerHTML={{
+                                __html: `${study.name}: ${study.description}`
+                            }}/>}
+                            maxLength={16}
+                        />
+                    </a>
+                );
+            }
+        }
+
+        return _renderColumnFn;
+    }
+
+    protected generateColumns() {
+
+        let visibleColumns = [
+            FusionTableColumnType.SAMPLE_ID,
+            FusionTableColumnType.SITE1_HUGO_SYMBOL,
+            FusionTableColumnType.SITE2_HUGO_SYMBOL,
+            FusionTableColumnType.TUMOR_READ_COUNT,
+            FusionTableColumnType.TUMOR_VARIANT_COUNT,
+            FusionTableColumnType.ANNOTATION
+        ];
+
+        this._columns = {};
+
+        if (this.props.studyIdToStudy) {
+            let studyObject = Object.assign({}, this.props.studyIdToStudy);
```

Comment:
theres no purpose for `studyObject`, just use Object.keys(this.props.studyIdToStudy)